mender-update/daemon/states.cpp

```diff
@@ -194,7 +194,10 @@ void UpdateDownloadState::OnEnter(Context &ctx, sm::EventPoster<StateEvent> &pos
 		[&poster](http::ExpectedIncomingResponsePtr exp_resp) {
 			if (!exp_resp) {
 				log::Error(exp_resp.error().String());
-				poster.PostEvent(StateEvent::Failure);
```

Comment:
Not sure I get it, but I'll have to trust you here

Reply:
The artifact reader also produces an error, which posts an event. So if we also post it here, it will be posted twice, most likely bringing the state machine into a bad state.